File: typescript/base-hmi-types/src/text-graphic-lists/HmiTextList.ts
import { IHmiObject } from "../IHmiObject.js";
import { HmiMultilingualText } from "../common/HmiMultilingualText.js";

export class HmiTextList implements IHmiObject {
  lastModified?: Date;
  name?: string;
  kind: HmiTextGraphicListKind = HmiTextGraphicListKind.Hmi;
  rangeType: HmiListRangeType = HmiListRangeType.Decimal;
  comment?: HmiMultilingualText;
  readonly entries: HmiTextListEntry[] = [];
}

export class HmiTextListEntry {
  name?: string;
  from = 0;
  to = 0;
  default = false;
  text?: HmiMultilingualText;
}

export enum HmiTextGraphicListKind {
  Hmi = "Hmi",
}

export enum HmiListRangeType {
  Decimal = "Decimal",
  Binary = "Binary",
  Bit = "Bit",
}

File: src/BaseHmiTypes/TextGraphicLists/HmiTextList.cs
using BaseHmiTypes.Common;

namespace BaseHmiTypes.TextGraphicLists;

public sealed class HmiTextList : IHmiObject
{
    public DateTime? LastModified { get; set; }

    public string? Name { get; set; }

    public HmiTextGraphicListKind Kind { get; set; } = HmiTextGraphicListKind.Hmi;

    public HmiListRangeType RangeType { get; set; }

    public HmiMultilingualText? Comment { get; set; }

    public IList<HmiTextListEntry> Entries { get; } = new List<HmiTextListEntry>();
}

public sealed class HmiTextListEntry
{
    public string? Name { get; set; }

    public int From { get; set; }

    public int To { get; set; }

    public bool Default { get; set; }

    public HmiMultilingualText? Text { get; set; }
}
